### PR TITLE
Added OS Specific Variable Files

### PR DESCRIPTION
**Pull request by DerekTBrown**
Created multiple variable files to expand OS support.

### CODE REVIEW

File: tasks/main.yml [hunk 1]
```diff
@@ -1,14 +1,23 @@
 ---
+
+- name: Include Configuration
+  include_vars: "{{ item }}"
+  with_first_found:
+    - "{{ ansible_os_family }}.yml"
+  when: etcd_config_only = false
```
Comment (retr0h) [inline tasks/main.yml]: `when: etcd_config_only == false` ?

File: tasks/main.yml [hunk 2]
```diff
@@ -17,19 +26,20 @@
...
+  when: etcd_daemon = "init"
   notify:
     - Restart etcd
 
 - name: Add etcd systemd service
   template: src=etc/systemd/system/etcd.j2
             dest=/etc/systemd/system/etcd.service
             mode=0644
-  when: ansible_os_family == "RedHat"
+  when: etcd_daemon = "init"
```
Comment (retr0h) [inline tasks/main.yml]: `when etcd_daemon == "systemctl"`, right?

File: tasks/main.yml (hunk 1 above)
Comment (DerekTBrown) [inline tasks/main.yml]: Fixed.

File: tasks/main.yml [hunk 3]
```diff
@@ -40,6 +40,6 @@
   template: src=etc/systemd/system/etcd.j2
             dest=/etc/systemd/system/etcd.service
             mode=0644
-  when: etcd_daemon = "init"
+  when: etcd_daemon = "systemctl"
```
Comment (retr0h) [inline tasks/main.yml]: Don't we need `==` not `=`?

**Comment by retr0h** [comment]
@akhoury6 can you verify this patch works as expected on your EL based systems?

File: tasks/main.yml [hunk 4]
```diff
@@ -17,19 +26,20 @@
   with_items:
     - etcd
     - etcdctl
+  when: etcd_config_only = false
```
Comment (retr0h) [inline tasks/main.yml]: Need to update this one too.

File: tasks/main.yml [hunk 5]
```diff
@@ -1,14 +1,23 @@
 ---
+
+- name: Include Configuration
+  include_vars: "{{ item }}"
+  with_first_found:
+    - "{{ ansible_os_family }}.yml"
+  when: not etcd_config_only
```
Comment (retr0h) [inline tasks/main.yml]: We should prob update all of these to `when: not etcd_config_only|bool`, so ansible properly handles [yes, no, true, false etc..]

**Comment by retr0h** [comment]
Molecule tests look good on trusty and centos/7.

```
$ molecule test
...
PLAY RECAP ********************************************************************
etcd-01                    : ok=9    changed=5    unreachable=0    failed=0
etcd-02                    : ok=9    changed=5    unreachable=0    failed=0
etcd-03                    : ok=9    changed=5    unreachable=0    failed=0

Idempotence test in progress (can take a few minutes)...
Idempotence test passed.


Executing testinfra tests found in tests/.
============================= test session starts ==============================
platform darwin -- Python 2.7.11, pytest-2.9.1, py-1.4.31, pluggy-0.3.1
rootdir: /Users/jodewey/git/ansible-etcd, inifile:
plugins: xdist-1.14, testinfra-1.1.2
gw0 [0] / gw1 [0] / gw2 [0]
scheduling tests via LoadScheduling

========================= no tests ran in 0.99 seconds =========================
```

```
$ molecule test --platform centos7
...
PLAY RECAP ********************************************************************
etcd-01                    : ok=8    changed=5    unreachable=0    failed=0
etcd-02                    : ok=8    changed=5    unreachable=0    failed=0
etcd-03                    : ok=8    changed=5    unreachable=0    failed=0

Idempotence test in progress (can take a few minutes)...
Idempotence test passed.


Executing testinfra tests found in tests/.
============================= test session starts ==============================
platform darwin -- Python 2.7.11, pytest-2.9.1, py-1.4.31, pluggy-0.3.1
rootdir: /Users/jodewey/git/ansible-etcd, inifile:
plugins: xdist-1.14, testinfra-1.1.2
gw0 [9] / gw1 [9] / gw2 [9]
scheduling tests via LoadScheduling
.........
========================== 9 passed in 10.20 seconds ===========================

No serverspec tests found in spec/.
```

**Pull request merged**